web/secondary_toolbar.js

```diff
@@ -76,6 +77,16 @@ class SecondaryToolbar {
         eventDetails: { tool: CursorTool.SELECT, }, close: true, },
       { element: options.cursorHandToolButton, eventName: 'switchcursortool',
         eventDetails: { tool: CursorTool.HAND, }, close: true, },
+      { element: options.scrollVerticalButton, eventName: 'switchscrollmode',
+        eventDetails: { mode: ScrollMode.VERTICAL, }, close: true, },
+      { element: options.scrollHorizontalButton, eventName: 'switchscrollmode',
+        eventDetails: { mode: ScrollMode.HORIZONTAL, }, close: true, },
+      { element: options.scrollWrappedButton, eventName: 'switchscrollmode',
+        eventDetails: { mode: ScrollMode.WRAPPED, }, close: true, },
+      { element: options.spreadOddButton, eventName: 'togglespreadmode',
+        eventDetails: { mode: SpreadMode.ODD, }, close: true, },
+      { element: options.spreadEvenButton, eventName: 'togglespreadmode',
+        eventDetails: { mode: SpreadMode.EVEN, }, close: true, },
```

Comment:
Okay, that makes sense to me! Just to be clear, this applies to all five new menu items (scroll modes and spread modes), and not just the spread modes?

Reply:
All of them.

Reply:
Hi - can I comment here please - and would be interested in others too - before you make a final decision here.
I believe the the menu should auto close for those modes - as isn't it better to treat the user as if they know what they want and close the menu for them. Rather than coming from the perspective that the user could have made a mistake and lets leave the menu open so it is easy for them to change back - wouldn't this be catering for a small percentage at the expense of the many.
It does make sense to me to not auto close for Rotation as that is in my mind more about changing the UI in a what does it look like manner.
I hope you will take this into account.
@Rob--W - many thanks for reviewing this PR - there are many of us testing this and looking forward to it being included.
Darren

Reply:
One difference that could be relevant (in support of continuing to let these buttons close the menu) between rotation and these view modes is that if a user wanted to invert their PDF, they'd have to click a rotation button twice, whereas (assuming no mistakes) all these new buttons only need to be clicked once. On the other hand (in support of preventing these buttons from closing the menu), perhaps users will frequently want to change scroll mode and set a spread lock at the same time. So I could go either way, personally.

Reply:
I understand what your saying re leaving the menu open if changing spread lock - but can't you do this selectively in the code (ie. don't auto close for Spread Odd or Even - but close for changing scroll mode.

Reply:
Absolutely, and that would handle the people who prefer to set spread first and scroll second, but not the people who do it the other way around. :smile: Per Rob--W, I'm holding off on making this change for now.